### ISSUE TITLE
Mod OptiFabric v1.11.9/v1.11.17 + preview OptiFine 1.17/1.17.1 pre25/26/27 causes incompatibility with the mod FancyMenu or library Konkrete

### ISSUE THREAD

**Issue by Disguys** [Keksuccino/FancyMenu#184]
**Describe the bug**
Mod OptiFabric v1.11.9 + preview OptiFine 1.17 pre25 causes incompatibility with the mod FancyMenu or library Konkrete.

**To Reproduce**
1. Install [OptiFabric v1.11.9](https://www.curseforge.com/minecraft/mc-mods/optifabric/files) and preview OptiFine 1.17 pre25
2. Install FancyMenu v2.1.2 and library Konkrete v1.2.1
3. Run fabric 1.17 and understand that this causes the crash of the game, without the FancyMenu mod and the Konkrete library, Minecraft with OptiFabric starts without problems.

**Expected behavior**
They were compatible with each other.
On version 1.16.5, there were no similar problems with OptiFabric and FancyMenu.

**Crash Log**
```[19:10:43] [main/INFO]: Loading for game Minecraft 1.17
[19:10:43] [ForkJoinPool-1-worker-3/WARN]: Non-Fabric mod JAR at "C:\Users\Disguys\curseforge\minecraft\Instances\DMP\mods\preview_OptiFine_1.17_HD_U_G9_pre25.jar", ignoring
[19:10:43] [main/INFO]: [FabricLoader] Loading 51 mods:
	- fabric@0.36.0+1.17
	- fabric-api-base@0.3.0+a02b44633d
	- fabric-api-lookup-api-v1@1.2.0+2b5c62d03d
	- fabric-biome-api-v1@3.1.11+c345aea83d
	- fabric-blockrenderlayer-v1@1.1.5+a02b44633d
	- fabric-command-api-v1@1.1.1+a02b44633d
	- fabric-commands-v0@0.2.2+92519afa3d
	- fabric-containers-v0@0.1.12+a02b44633d
	- fabric-content-registries-v0@0.2.2+a02b44633d
	- fabric-crash-report-info-v1@0.1.5+be9da3103d
	- fabric-dimensions-v1@2.0.10+a02b44633d
	- fabric-entity-events-v1@1.1.0+a02b44633d
	- fabric-events-interaction-v0@0.4.9+a722d8c03d
	- fabric-events-lifecycle-v0@0.2.1+92519afa3d
	- fabric-game-rule-api-v1@1.0.6+a02b44633d
	- fabric-item-api-v1@1.2.4+a02b44633d
	- fabric-item-groups-v0@0.2.10+b7ab61213d
	- fabric-key-binding-api-v1@1.0.4+a02b44633d
	- fabric-keybindings-v0@0.2.2+36b77c3e3d
	- fabric-lifecycle-events-v1@1.4.4+a02b44633d
	- fabric-loot-tables-v1@1.0.4+a02b44633d
	- fabric-mining-levels-v0@0.1.3+92519afa3d
	- fabric-models-v0@0.3.0+a02b44633d
	- fabric-networking-api-v1@1.0.11+b7ab61213d
	- fabric-networking-blockentity-v0@0.2.11+a02b44633d
	- fabric-networking-v0@0.3.2+92519afa3d
	- fabric-object-builder-api-v1@1.10.9+b7ab61213d
	- fabric-object-builders-v0@0.7.3+a02b44633d
	- fabric-particles-v1@0.2.4+a02b44633d
	- fabric-registry-sync-v0@0.7.10+e2961fee3d
	- fabric-renderer-api-v1@0.4.3+676f40fa3d
	- fabric-renderer-indigo@0.4.8+a02b44633d
	- fabric-renderer-registries-v1@3.2.0+a02b44633d
	- fabric-rendering-data-attachment-v1@0.1.5+a02b44633d
	- fabric-rendering-fluids-v1@0.1.13+a02b44633d
	- fabric-rendering-v0@1.1.2+92519afa3d
	- fabric-rendering-v1@1.6.0+a02b44633d
	- fabric-resource-loader-v0@0.4.7+b7ab61213d
	- fabric-screen-api-v1@1.0.4+155f865c3d
	- fabric-screen-handler-api-v1@1.1.8+a02b44633d
	- fabric-structure-api-v1@1.1.11+e36969833d
	- fabric-tag-extensions-v0@1.1.4+a02b44633d
	- fabric-textures-v0@1.0.6+a02b44633d
	- fabric-tool-attribute-api-v1@1.2.12+b7ab61213d
	- fabricloader@0.11.6
	- fancymenu@2.1.2
	- java@16
	- konkrete@1.2.1
	- minecraft@1.17
	- mm@2.2
	- optifabric@1.11.9
[19:10:44] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.2 Source=file:/C:/Users/Disguys/curseforge/minecraft/Install/libraries/net/fabricmc/sponge-mixin/0.9.4+mixin.0.8.2/sponge-mixin-0.9.4+mixin.0.8.2.jar Service=Knot/Fabric Env=CLIENT
[19:10:44] [main/INFO]: Compatibility level set to JAVA_16
[19:10:45] [main/WARN]: @Mixin target net/minecraft/class_2474$class_5124 is public in fabric-tag-extensions-v0.mixins.json:MixinObjectBuilder and should be specified in value
[19:10:47] [main/WARN]: Failed to add PDH Counter: \Файл подкачки(_Total)\% Usage, Error code: 0xC0000BB8
[19:10:47] [main/WARN]: Failed to add counter for PDH object: Файл подкачки
[19:10:47] [main/WARN]: Disabling further attempts to query Paging File.
[19:10:48] [main/INFO]: [OptiFine] (Reflector) Field not found: World.tileEntitiesToBeRemoved
[19:10:49] [main/FATAL]: Mixin apply failed konkrete.mixins.events.client.json:MixinGameRenderer -> net.minecraft.class_757: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException Implicit variable modifier injection failed in net/minecraft/class_757::onMatrixStackCreation [INJECT Applicator Phase -> konkrete.mixins.events.client.json:MixinGameRenderer -> Apply Injections ->  -> Inject -> konkrete.mixins.events.client.json:MixinGameRenderer->@ModifyVariable::onMatrixStackCreation(Lnet/minecraft/class_4587;)Lnet/minecraft/class_4587;]
org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Implicit variable modifier injection failed in net/minecraft/class_757::onMatrixStackCreation [INJECT Applicator Phase -> konkrete.mixins.events.client.json:MixinGameRenderer -> Apply Injections ->  -> Inject -> konkrete.mixins.events.client.json:MixinGameRenderer->@ModifyVariable::onMatrixStackCreation(Lnet/minecraft/class_4587;)Lnet/minecraft/class_4587;]
	at org.spongepowered.asm.mixin.injection.modify.ModifyVariableInjector.inject(ModifyVariableInjector.java:169) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.injection.code.Injector.inject(Injector.java:264) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.inject(InjectionInfo.java:385) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1284) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1042) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569) [sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351) [sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208) [sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178) [sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23) [fabric-loader-0.11.6.jar:0.9.4+mixin.0.8.2]
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:162) [fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:154) [fabric-loader-0.11.6.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:519) [?:?]
	at java.lang.Class.getDeclaredFields0(Native Method) ~[?:?]
	at java.lang.Class.privateGetDeclaredFields(Class.java:3229) [?:?]
	at java.lang.Class.getDeclaredFields(Class.java:2335) [?:?]
	at net.optifine.reflect.FieldLocatorTypes.<init>(FieldLocatorTypes.java:25) [Optifine-mapped.jar:?]
	at net.optifine.reflect.Reflector.<clinit>(Reflector.java:518) [Optifine-mapped.jar:?]
	at net.minecraft.class_128.<init>(class_128.java:44) [intermediary-1.17.jar:?]
	at net.minecraft.class_128.method_24305(class_128.java:349) [intermediary-1.17.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:137) [intermediary-1.17.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234) [fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153) [fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28) [fabric-loader-0.11.6.jar:?]
Caused by: org.spongepowered.asm.mixin.injection.modify.InvalidImplicitDiscriminatorException: Found 2 candidate variables but exactly 1 is required.
	at org.spongepowered.asm.mixin.injection.modify.LocalVariableDiscriminator.findImplicitLocal(LocalVariableDiscriminator.java:349) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.injection.modify.LocalVariableDiscriminator.findLocal(LocalVariableDiscriminator.java:320) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	at org.spongepowered.asm.mixin.injection.modify.ModifyVariableInjector.inject(ModifyVariableInjector.java:160) ~[sponge-mixin-0.9.4+mixin.0.8.2.jar:0.9.4+mixin.0.8.2]
	... 30 more
```

**Basic Informations (please complete the following information):**
 - OS: Windows 10 Pro 64 bit
 - FancyMenu Version 2.1.2
 - Fabric Version 0.11.6
 - Minecraft Version 1.17
 - Active Mods: FabricAPI v0.36.0, FancyMenu v2.1.2, Konkrete v1.2.1, OptiFabric v1.11.9, preview OptiFine 1.17 HD U G9 pre25.

**Comment by Disguys** [comment]
Still not compatible with OptiFabric v1.11.17 and preview OptiFine pre27 for Minecraft 1.17.1

**Comment by Keksuccino** [comment]
Yes I know that, because I haven't fixed it yet.

**Comment by Keksuccino** [comment]
Fixed in Konkrete v1.2.2! Thank you for your report!
